### PR TITLE
2372 document terraform

### LINKED ISSUE #2372: Spike: Quick zoo setup
# Spike

## Objective
We need to be able to quickly and easily create monkey zoo environments.

- [x] Setup process is well-documented
- [x] Setup is quick and easy

## Scope
Estimated 0d @ilija-lazoroski 

## Output
Add tasks to #2366

## Tasks Remaining

- [ ] Ensure machines are configured to have external IPs. (0.25d) @ilija-lazoroski 
- [ ] Ensure the resources (cpu,memory,etc) for each machine in the terraform scripts matches what's in the guardicore-monkey environment (0.25d) @ilija-lazoroski

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes #2372 .

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: envs/monkey_zoo/docs/zoo_setup.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+
+
+## Getting started:
+
+Requirements:
+1.  Have terraform installed (for installation instructions check this
+    [link](https://learn.hashicorp.com/tutorials/terraform/install-cli)).
+2.  Have a Google Cloud Platform account (upgraded if you want to test
+    whole network at once).
```
Comment (VakarisZ) [inline envs/monkey_zoo/docs/zoo_setup.md:11]: Gcloud cli is required as well?

Comment (VakarisZ) [reply envs/monkey_zoo/docs/zoo_setup.md:11]: Do we need to specify that `envs/monkey_zoo` folder needs to be downloaded and all actions are done in reference to it? These files can be viewed online in git

File: envs/monkey_zoo/docs/zoo_setup.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+           monkeyzoo_project="guardicore-22050661" // Project where monkeyzoo images are kept. Leave as is.
+
+         }
+
+4.  Run terraform init
+
+To deploy the network run:<br>
+`terraform plan` (review the changes it will make on GCP)<br>
+`terraform apply` (adds machines to these networks)
```
Comment (VakarisZ) [inline envs/monkey_zoo/docs/zoo_setup.md:66]: Was one apply enough? Last time I had to do it twice, one to create networks and another one to add machines

File: envs/monkey_zoo/docs/zoo_setup.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+### How to get into the islands:
+
+**island-linux-250:** SSH from GCP
+
+**island-windows-251:** In GCP/VM instances page click on
+island-windows-251. Set password for your account and then RDP into
+the island.
+
+## These are most common steps on monkey islands:
```
Comment (VakarisZ) [inline envs/monkey_zoo/docs/zoo_setup.md:79]: These are not steps, you're just explaining where you should install the island.

File: envs/monkey_zoo/docs/zoo_setup.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+### For users
+
+Upload the AppImage deployment option and run it in island-linux-250.
+Or upload the MSI deployment option, install it and run it in island-windows-251.
+After that use the Monkey as you would on local network.
```
Comment (VakarisZ) [inline envs/monkey_zoo/docs/zoo_setup.md:85]: Call this section "Installing the Infection Monkey" and just note which machines should be used and maybe how you can access them (you need to create a password for rdp). Then just drop a link to our documentation about setup

File: envs/monkey_zoo/docs/zoo_setup.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+Execute C:\\run\_monkey.bat as administrator
+
+To update repository:<br>
+1\. Open cmd as an administrator<br>
+2\. `cd C:\infection_monkey`<br>
+3\. `git pull` (updates develop branch)<br>
+
+Update all requirements using deployment script:<br>
+1\. `cd C:\infection_monkey\deployment_scripts`<br>
+2\. `./run_script.bat "C:\infection_monkey" "develop"`<br>
```
Comment (VakarisZ) [inline envs/monkey_zoo/docs/zoo_setup.md:119]: This is not the place for documentation about running the island from source (it has nothing to do with the zoo)

File: envs/monkey_zoo/docs/zoo_setup.md (hunk 3 above)
Comment (VakarisZ) [reply envs/monkey_zoo/docs/zoo_setup.md:79]: We should explain in this section how you can use the zoo. For example how do you actually exploit machines? What credentials need to be used to successfully exploit ssh machine for example. None of that is mentioned

**Review by VakarisZ — CHANGES_REQUESTED** [review]
We need to elaborate on how to actually craft configs and use more links to our current documentation and to the `zoo_network.md`.

File: envs/monkey_zoo/docs/zoo_setup.md (hunk 2 above)
Comment (ilija-lazoroski) [reply envs/monkey_zoo/docs/zoo_setup.md:66]: Just one was enough

File: envs/monkey_zoo/docs/zoo_setup.md (hunk 5 above)
Comment (ilija-lazoroski) [reply envs/monkey_zoo/docs/zoo_setup.md:119]: This has everything to do with the zoo. This is explaining how we can run the island from source on the zoo machine. It is tested and it works.

File: envs/monkey_zoo/docs/zoo_setup.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
+# Setting up a Monkey Zoo using Terraform
+
+## Getting started:
+
+Requirements:
+1.  Have terraform installed (for installation instructions check this
+    [link](https://learn.hashicorp.com/tutorials/terraform/install-cli)).
+2.  Have a Google Cloud Platform account (upgraded if you want to test
+    whole network at once).
+3.  Have gcloud installed (for installation instruction check this
```
Comment (VakarisZ) [inline envs/monkey_zoo/docs/zoo_setup.md:10]:
```suggestion
3.  Have gcloud CLI installed (for installation instruction check this
```

File: envs/monkey_zoo/docs/zoo_setup.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+#### Linux
+
+Upload the [`AppImage deployment`](https://github.com/guardicore/monkey/releases/latest)
+option and run it in [`island-linux-250`](zoo_network.md#_Toc536021489).
+
+#### Windows
+
+Upload the [`MSI deployment`](https://github.com/guardicore/monkey/releases/latest)
+option, install it and run it in [`island-windows-251`](zoo_network.md#_Toc536021490).
+After that use the Monkey as you would on local network.
```
Comment (VakarisZ) [inline envs/monkey_zoo/docs/zoo_setup.md:101]: This whole section should probably be simplified to "Login to `island-linux-250` or `island-windows-251` depending on your preferred operating system. Then follow the [setup guide](https://techdocs.akamai.com/infection-monkey/docs/setting-up-infection-monkey)"

File: envs/monkey_zoo/docs/zoo_setup.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+
+### For developers
```
Comment (VakarisZ) [inline envs/monkey_zoo/docs/zoo_setup.md:104]: Just link to https://techdocs.akamai.com/infection-monkey/docs/development-setup ?

Comment (ilija-lazoroski) [reply envs/monkey_zoo/docs/zoo_setup.md:104]: The island machines in the zoo has a Monkey repo already in there which either we should delete or we should explain how we can use it. This section explains on how to use that repo in the zoo islands.

**Review by VakarisZ — APPROVED** [review]

**Pull request merged**